posthog/hogql_queries/insights/lifecycle_query_runner.py

```diff
@@ -100,7 +100,13 @@ def to_persons_query(
                     ast.CompareOperation(
                         op=ast.CompareOperationOp.Eq,
                         left=ast.Field(chain=["start_of_period"]),
-                        right=ast.Constant(value=day),
+                        right=parse_expr(
+                            "dateTrunc({interval}, toDateTime({day}))",
```

Comment:
Just FYI, I've seen `date_truct` give inconsistent results compared to using `toStartOf<Interval>`. Meaning I had to switch all of trends query runner to use `toStartOf..` instead of a mix between the two. If you're using both within lifecycles, it may be best to switch to one or the other depending on the use case here